src/geometry/ArborX_DetailsAlgorithms.hpp

```diff
@@ -573,6 +574,165 @@ struct intersects<PointTag, TriangleTag, Point, Triangle>
   }
 };
 
+template <typename Box, typename Triangle>
+struct intersects<BoxTag, TriangleTag, Box, Triangle>
+{
+  KOKKOS_FUNCTION static constexpr bool apply(Box const &box,
+                                              Triangle const &triangle)
+  {
+    // Based on the Separating Axis Theorem
+
+    // Test bounding boxes, i.e., normals of the box
+    Box triangle_bounding_box;
+    ArborX::Details::expand(triangle_bounding_box, triangle);
+    if (!ArborX::Details::intersects(triangle_bounding_box, box))
+      return false;
+
+    // shift to origin
+    constexpr int DIM = GeometryTraits::dimension_v<Triangle>;
+    static_assert(DIM == 3,
+                  "Box-Triangle intersection only implemented in 3d!");
+    using Float = typename GeometryTraits::coordinate_type_t<Triangle>;
+    auto min_corner = box.minCorner();
+    auto max_corner = box.maxCorner();
+    auto triangle_a = triangle.a;
+    auto triangle_b = triangle.b;
+    auto triangle_c = triangle.c;
+    ExperimentalHyperGeometry::Point<DIM, Float> edge_ab;
+    ExperimentalHyperGeometry::Point<DIM, Float> edge_ac;
+    ExperimentalHyperGeometry::Point<DIM, Float> extents;
```

Comment:
Maybe introduce `Vector` as in distance point-triangle functor? I don't really like proliferation, though. We would probably want some common `Vector` functionality that is independent of `Ray` and is multi-dimensional.

Reply:
Sure, we can do in a follow-up.

Reply:
#1070